protected/controllers/AdminFileController.php

```diff
@@ -381,59 +380,53 @@ public function actionUpdate($id)
 
         // Uncomment the following line if AJAX validation is needed
         // $this->performAjaxValidation($model);
-        if (isset($_POST['edit_attr'])) {
-            $args = $_POST['FileAttributes']['edit'];
-            $fa = FileAttributes::model()->findByPk($args['id']);
-            if($fa) {
-                $fa->attribute_id = $args['attribute_id'];
-                $fa->value = $args['value'];
-                if($args['unit_id'])
-                    $fa->unit_id = $args['unit_id'];
-
-                if($fa->validate()) {
-                    if($fa->save())
-                        $this->redirect(array('update','id'=>$model->id));
-                    else
-                        Yii::log('save attr failed', 'debug');
-                } else
-                    Yii::log(print_r($fa->getErrors(), true), 'debug');
+        if (Yii::$app->request->post('edit_attr') || Yii::$app->request->post('submit_attr')) {
+            $action = Yii::$app->request->post('edit_attr') ? 'edit' : 'new';
+            $args = Yii::$app->request->post('FileAttributes')[$action];
+            $fa = 'edit' === $action ? FileAttributes::model()->findByPk($args['id']) : $attribute;
+
+            if (!$fa) {
+                throw new CHttpException(404, 'FileAttribute not found');
             }
-        }
-        elseif(isset($_POST['submit_attr'])) {
-            $attrs = $_POST['FileAttributes']['new'];
-            $attribute->attribute_id = $attrs['attribute_id'];
-            $attribute->value = $attrs['value'];
-            if($attrs['unit_id'])
-                $attribute->unit_id = $attrs['unit_id'];
-
-            if($attribute->validate()) {
-                $attribute->save();
-                $this->redirect(array('update', 'id' => $model->id));
+
+            $fa->attribute_id = $args['attribute_id'];
+            $fa->value = $args['value'];
+            if ($args['unit_id']) {
+                $fa->unit_id = $args['unit_id'];
             }
-        } elseif (isset($_POST['File'])) {
-            $model->attributes = $_POST['File'];
 
-            $model->setSizeValue();
-            if ($model->validate()) {
-                $model->save();
+            if ($fa->save()) {
+                return $this->redirect(array('update', 'id' => $model->id));
+            } else {
+                Yii::log(print_r($fa->getErrors(), true), 'debug');
+            }
+
+        } elseif ($attrs = Yii::$app->request->post('File')) {
+            $model->attributes = $attrs;
+
+            if(!$model->setSizeValue()) {
+                $model->addError('size', 'You need to indicate a valid size size for the file');
```

Comment:
Error message has 'size' written twice. Should be: `You need to indicate a valid size for the file`?